### PR TITLE
8318410: jdk/java/lang/instrument/BootClassPath/BootClassPathTest.sh fails on Japanese Windows

### PR DESCRIPTION
**Pull request by yukikimmura**
I would like to fix this issue
because the test dose not work as intended.
Could someone please review it? 

Thanks,
Kimura Yukihiro

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8318410](https://bugs.openjdk.org/browse/JDK-8318410): jdk/java/lang/instrument/BootClassPath/BootClassPathTest.sh fails on Japanese Windows (**Bug** - P4)

### Reviewers
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16235/head:pull/16235` \
`$ git checkout pull/16235`

Update a local copy of the PR: \
`$ git checkout pull/16235` \
`$ git pull https://git.openjdk.org/jdk.git pull/16235/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16235`

View PR using the GUI difftool: \
`$ git pr show -t 16235`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16235.diff">https://git.openjdk.org/jdk/pull/16235.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16235#issuecomment-1767924339)

### CODE REVIEW

File: test/jdk/java/lang/instrument/BootClassPath/Setup.java [hunk 1]
```diff
@@ -45,6 +45,10 @@ public static void main(String[] args) throws Exception {
         }
         String workDir = args[0];
         String premainClass = args[1];
+        boolean isCygwin = false;
+        if (args.length==3 && args[2].equals("CYGWIN")) {
```
Comment (sspitsyn) [inline test/jdk/java/lang/instrument/BootClassPath/Setup.java:49]: Nit: Could you, please, add spaces around `==`?

**Review by sspitsyn — APPROVED** [review]
Hi Kimura,

The fix looks okay to me. Thank you for taking care about it.
I'm not aware much about all these encoding names and their correctness.
How did you test all these languages?

Thanks,
Serguei

File: test/jdk/java/lang/instrument/BootClassPath/Setup.java (hunk 1 above)
Comment (yukikimmura) [reply test/jdk/java/lang/instrument/BootClassPath/Setup.java]: Thank you for your advice.  
I fixed it as you mentioned.

Thanks,
Kimura Yukihiro

**Comment by yukikimmura** [comment]
Hi Serguei,

Thank you for your comment. 

I tested these encoding name using LANG, LC_ALL and LC_CTYPE environment variables on Linux.
For instance, when I tested "ETC_TW", I set LANG and  LC_CTYPE to zh_TW.euctw which I got with "locale -a" command. 
I took these encoding names from 
GetJavaProperties() and ParseLocale() in java.base/unix/native/libjava/java_props_md.c.

And I tested only English and Japanese Windows because I could not make all environments. 
I think these names are correct because I took them from 
GetJavaProperties() and getEncodingInternal() in java.base/windows/native/libjava/java_props_md.c.

Thanks,
Kimura Yukihiro

**Comment by yukikimmura** [comment]
Hi Serguei,

Thank you for your quick reply.
Could you please sponsor the fix?

Thanks,
Kimura Yukihiro

**Comment by sspitsyn** [comment]
> Hi Serguei,
> 
> Thank you for your quick reply. Could you please sponsor the fix?
> 
> Thanks, Kimura Yukihiro

I was hoping to have one more reviewer of your fix.
But it seems there is nobody else to review it.
Sponsored now.

**Comment by yukikimmura** [comment]
This project has not yet received a notification from the bot that I can type the integrate command.
Should I adde a noreg-self label on JBS, or should I find one more reviewer?

Thanks,
Kimura Yukihiro

**Comment by sspitsyn** [comment]
> This project has not yet received a notification from the bot that I can type the integrate command. Should I adde a noreg-self label on JBS, or should I find one more reviewer?
> 
> Thanks, Kimura Yukihiro

You can see this message from bot:
> @sspitsyn The change author (@yukikimmura) must issue an integrate command before the integration can be sponsored.

So, please, go ahead and issue the command `/integrate` .
Then I'll sponsor this fix again.

**Comment by sspitsyn** [comment]
I've added the label `noreg-self`.

**Comment by kevinjwalls** [comment]
There is an "integration blocker" note above, the title of this PR needs updating: 
Title mismatch between PR and JBS for issue [JDK-8318410](https://bugs.openjdk.org/browse/JDK-8318410)

**Comment by yukikimmura** [comment]
/integrate

I am grateful for your support.

**Comment by sspitsyn** [comment]
Seems to be integrated now. :)

**Pull request closed without merging**